internal/ad/backends/winbind/winbind.go

```diff
@@ -110,8 +134,21 @@ func (w Winbind) Domain() string {
 }
 
 // HostKrb5CCNAME returns the absolute path of the machine krb5 ticket.
-func (w Winbind) HostKrb5CCNAME() string {
-	return w.hostKrb5CCNAME
+func (w Winbind) HostKrb5CCNAME() (string, error) {
+	target := "/tmp/krb5cc_0"
+
+	// Uppercase domain and hostname, and trim domain from hostname if needed
+	domain := strings.ToUpper(w.domain)
+	hostname := strings.TrimSuffix(strings.ToUpper(w.hostname), "."+domain)
+
+	principal := fmt.Sprintf("%s$@%s", hostname, domain)
+	cmdArgs := append(w.kinitCmd, "-k", principal, "-c", target)
+	if cmd, err := exec.Command(cmdArgs[0], cmdArgs[1:]...).CombinedOutput(); err != nil {
```

Comment:
I'm pretty sure the smbsafe mutex stuff, that I've mentioned to you about the cmd executions, applies here as well.